### DISCUSSION TITLE
Ability to Connect to Dev Server via USB on iOS

### DISCUSSION THREAD

**Discussion by peetzweg** [react-native-community/discussions-and-proposals#374] [Ideas]
![IMG_552137CB5CE4-1](https://user-images.githubusercontent.com/839848/81392548-23459d80-911f-11ea-863e-270cb5fdf329.jpeg)

### Introduction

Enable RN debugging on iOS via USB.

### The Core of It

From the docs: (https://reactnative.dev/docs/running-on-device.html#connecting-to-the-development-server )
`You only have to be on the same Wi-Fi network as your computer.` 

Yes, that's super convenient, however, I recently started to develop an App that needs to change the current Wifi of the device. Thereby losing the connection to the Development server. This results in a continuous error message on the screen about the lost connection to the dev server. Disabling "remote debugging" helps, although that's not what I want while developing.

On Android, the Apps connection to the dev server is handled via USB. Can we somehow implement this on iOS as well? The phone is often connected to the Computer via USB anyways to install new builds of the native code.

FYI: Using the simulator is not an option as working with `NEHotspotConfiguration` throws an exception which indicates wrong entitlements. Which are set correctly though, as it works on a physical device. Related issue https://github.com/JuanSeBestia/react-native-wifi-reborn/issues/71

FYI2: Tried using Flipper without success.

### Discussion points

+ is there an actual need for USB debugging on iOS or is there another workaround I could use?
+ Wifi by default, USB by choice?
+ Make "Runtime is not ready for debugging. Make Sure Packager Server is running" dismissable.

**Comment by mreed49** [comment]
I have the same exact use case as you. I've resorted to building a release version of the app with every change. Pretty brutal lol. 

Testing any sort of network condition on a physical iOS device is pretty hard to do with the current react native environment.

**Comment by ChaossAdept7** [comment]
I also need to do something with changing networks, but.. No luck with it.

**Comment by AndreiCalazans** [comment]
I think this is a valid request. Where would this need to be implemented though? In the CLI? Or in core?

**Comment by wanglaohushiwo** [comment]
> I have the same exact use case as you. I've resorted to building a release version of the app with every change. Pretty brutal lol.
> 
> Testing any sort of network condition on a physical iOS device is pretty hard to do with the current react native environment.

There is a solution to this problem, https://github.com/olofd/react-native-FBPortForwarding。But the version of react is too old。

**Comment by wanglaohushiwo** [comment]
> I also need to do something with changing networks, but.. No luck with it.

There is a solution to this problem, https://github.com/olofd/react-native-FBPortForwarding。But the version of react is too old。

**Comment by wanglaohushiwo** [comment]
There is a solution to this problem, https://github.com/olofd/react-native-FBPortForwarding。But the version of react is too old。
